dotnet/Simple.Sauce/SauceOptions.cs

```diff
@@ -56,19 +149,86 @@ public SauceOptions()
         public string TimeZone { get; set; }
         public string TunnelIdentifier { get; set; }
         public bool VideoUploadOnPass { get; set; }
+        public DriverOptions SeleniumOptions { get; set; }
+        public Browser BrowserName { get; set; } = Browser.Chrome;
 
-        public void WithEdge()
+        public DriverOptions ToDriverOptions()
         {
-            WithEdge(EdgeVersion.Latest);
+            //TODO temporary solution to get the code working
+            var sauceConfiguration = new Dictionary<string, object>
+            {
+                ["username"] = Environment.GetEnvironmentVariable("SAUCE_USERNAME"),
+                ["accessKey"] = Environment.GetEnvironmentVariable("SAUCE_ACCESS_KEY")
+            };
+            //TODO add if logic from the toCapabilities() in Java
+
+            if (BrowserName == Browser.Edge)
+                SeleniumOptions = new EdgeOptions();
+            else if (BrowserName == Browser.Firefox)
+                SeleniumOptions = new FirefoxOptions();
+            else
+                throw new ArgumentOutOfRangeException("The desired browser configuration is not yet set.");
+
+
+            W3CAllowedOptionsList.ForEach(AppendCapabilityToSeleniumOptions);
+
+            SeleniumOptions.AddAdditionalOption("sauce:options", sauceConfiguration);
+            return SeleniumOptions;
+        }
+
+        private void AppendCapabilityToSeleniumOptions(string capability)
+        {
+            var capabilityValue = TryToGetCapabilityValue(capability);
+            if (capabilityValue != null)
+                SeleniumOptions.AddAdditionalOption(capability, capabilityValue);
+        }
+
+        private object TryToGetCapabilityValue(string capability)
+        {
+            try
+            {
+                return GetType().GetProperty(capability)?.GetValue(this);
+            }
+            catch (NullReferenceException)
+            {
+                return null;
+            }
+
+            //PropertyInfo pinfo = typeof(SauceOptions).GetProperty(capability);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/SauceLabs/simple_sauce/pullRequest?prid=4945851)